tests/integration/test_remoted/test_agent_communication/test_agent_pending_status.py

```diff
@@ -0,0 +1,115 @@
+from wazuh_testing.utils.callbacks import generate_callback
+from wazuh_testing.tools.thread_executor import ThreadExecutor
+from . import CONFIGS_PATH, TEST_CASES_PATH
+
+
+
+# Set pytest marks.
+pytestmark = [pytest.mark.server, pytest.mark.tier(level=1)]
+
+# Cases metadata and its ids.
+cases_path = Path(TEST_CASES_PATH, 'cases_agent_pending_status.yaml')
+config_path = Path(CONFIGS_PATH, 'config_agent_pending_status.yaml')
+test_configuration, test_metadata, cases_ids = get_test_cases_data(cases_path)
+test_configuration = load_configuration_template(config_path, test_configuration, test_metadata)
+
+daemons_handler_configuration = {'all_daemons': True}
+
+local_internal_options = {REMOTED_DEBUG: '2'}
+
+injectors = []
+
+
+def send_event(protocol, manager_port, agent):
+    """Send an event to the manager"""
+
+    sender, injector = connect(agent, manager_port = manager_port, protocol = protocol, wait_status='pending')
+    injectors.append(injector)
+    return injector
+
+# Test function.
+@pytest.mark.parametrize('test_configuration, test_metadata',  zip(test_configuration, test_metadata), ids=cases_ids)
+def test_agent_pending_status(test_configuration, test_metadata, configure_local_internal_options, truncate_monitored_files,
+                            set_wazuh_configuration, daemons_handler, simulate_agents):
+
+    '''
+    description: Validate agent status after sending only the start-up
+
+    parameters:
+        - test_configuration
+            type: dict
+            brief: Configuration applied to ossec.conf.
+        - test_metadata:
+            type: dict
+            brief: Test case metadata.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate all the log files and json alerts files before and after the test execution.
+        - configure_local_internal_options:
+            type: fixture
+            brief: Configure the Wazuh local internal options using the values from `local_internal_options`.
+        - daemons_handler:
+            type: fixture
+            brief: Restart service once the test finishes stops the daemons.
+        - simulate_agents
+            type: fixture
+            brief: create agents
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Apply changes to the ossec.conf configuration.
+
+    '''
+
+    log_monitor = FileMonitor(WAZUH_LOG_PATH)
+
+
+    agents = simulate_agents
+
+    log_monitor.start(callback=generate_callback(KEY_UPDATE))
+    assert log_monitor.callback_result
+
+
+    manager_port = test_metadata['port']
+    protocol = test_metadata['protocol']
+    send_event_threads = []
+
+
+    for agent in agents:
+
+        # Create sender event threads
+        send_event_threads.append(ThreadExecutor(send_event, {'protocol': protocol,
+                                                            'manager_port': manager_port, 'agent': agent}))
+
+    # Wait 10 seconds until remoted is fully initialized
```

Comment:
Can you do this with a fixture?

Reply:
I spoke with Tomas and an agreement was reached not to do it due to how the tests were designed.